config/routes.rb

```diff
@@ -1,7 +1,9 @@
 Rails.application.routes.draw do
   root to: 'users#index'
-  resources :users, only: %i[index show] do
-    resources :posts, only: %i[index show]
+  resources :users do
+    resources :posts do
+      resources :comments, :likes
```

Comment:
- [ ] [OPTIONAL] Please consider exposing only the needed routes. For instance, here we only need the `create` action for the `likes`. And for the `comments`, we need only the `GET` and `POST` routes 👍